File: vercel-app/scripts/tmp_scan_i18n_kh_vi_ms.mjs
import fs from "node:fs"

const text = fs.readFileSync("lib/i18n.ts", "utf8")

function block(name, nxt) {
  const s = text.indexOf(`  ${name}: {`)
  const e = text.indexOf(`  ${nxt}: {`)
  if (s < 0 || e < 0) throw new Error(`missing ${name} or ${nxt}`)
  return text.slice(s, e)
}

for (const [lang, nxt] of [
  ["kh", "vi"],
  ["vi", "ms"],
  ["ms", "} as const"],
]) {
  const b = block(lang, nxt)
  const keys = [...b.matchAll(/^\s{4}((?:posTour|posDemo|posMainTour)\w*):/gm)]
  console.log(lang, "chars", b.length, "demo_tour_keys", keys.length)
}

File: vercel-app/scripts/tmp_scan_i18n_kh_vi_ms.py
# -*- coding: utf-8 -*-
"""One-off: scan kh/vi/ms blocks in lib/i18n.ts for posTour* keys count."""
import re
from pathlib import Path

text = Path("lib/i18n.ts").read_text(encoding="utf-8")


def block(name: str, nxt: str) -> str:
    s = text.find(f"  {name}: {{")
    e = text.find(f"  {nxt}: {{")
    if s < 0 or e < 0:
        raise SystemExit(f"missing {name} or {nxt}")
    return text[s:e]


for lang, nxt in [("kh", "vi"), ("vi", "ms"), ("ms", "} as const")]:
    b = block(lang, nxt)
    keys = re.findall(r"^\s{4}((?:posTour|posDemo|posMainTour)\w*):", b, re.M)
    print(lang, "chars", len(b), "demo_tour_keys", len(keys))
